Update Ariadne to 0.5.1 and fix examples

bytexenon opened the pull request:
When trying to use the code snippets from the examples with a more recent version of `ariadne`, I encountered compilation errors due to API changes between Ariadne v0.2 and later versions (specifically v0.3+). This can be confusing for users trying to learn the library or adapt the examples. Keeping the examples aligned with a recent version of `ariadne` improves the developer experience.

This also addresses issue #714, which requested updating the examples for Ariadne 0.5.

**Changes:**

*   Updated `ariadne` in `Cargo.toml` (dev-dependencies) and `Cargo.lock` to `0.5.1`.
*   Modified all examples (`examples/*.rs`) to use the updated `ariadne` API, specifically:
    *   `Report::build` now takes `(SourceId, Span)` instead of `SourceId, Offset`.
    *   `Label::new` now takes `(SourceId, Span)` instead of just `Span`.

I've confirmed that the examples build and run correctly with these changes.

Fixes #714

zesterer commented:
Was this closed for a particular reason? Apologies about not getting to it sooner.

bytexenon commented:
> Was this closed for a particular reason? Apologies about not getting to it 

Hi! No worries about the delay. I saw that PR #760 got merged, which also updated the examples to use Ariadne 0.5. Since that accomplished the same goal, I deleted the branch for my PR as it was no longer necessary

zesterer commented:
Thanks, I think I'd confused the two PRs!
